low-level/files/opt/seagate/sspl/setup/sspl_reset.py

```diff
@@ -0,0 +1,178 @@
+
+SSPL_CONFIGURED=f"/var/{PRODUCT_FAMILY}/sspl/sspl-configured"
+
+class SSPLResetCmd:
+
+    name="sspl-reset"
+
+    def __init__(self, args: dict):
+        self._args = args.args
+        self._config()
+
+    def _config(self):
+        if os.path.exists(file_store_config_path):
+            with open(file_store_config_path, "r") as SSPL_CONF:
+                for line in SSPL_CONF:
+                    if 'data_path' in line:
+                        self.SSPL_DATA_DIR=line.split("=")[1]
+                    elif 'log_file_path' in line:
+                        self.IEM_FILE_PATH=line.split("=")[1]
+        else:
+            self.SSPL_DATA_DIR=f"/var/{PRODUCT_FAMILY}/sspl/data"
+            self.IEM_FILE_PATH=f"/var/{PRODUCT_FAMILY}/sspl/data/iem/iem_messages"
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @staticmethod
+    def usage(prog: str):
+        """Print usage instructions"""
+
+        sys.stderr.write(
+            f"usage: {prog} [hard [-p <LDR_R1>]|soft]\n"
+            f"hard options:\n"
+            f"-p product to be configured\n")
+    
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """Return the Command after parsing the command line."""
+
+        parser = argparse.ArgumentParser(desc)
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+
+        args, unknown = parser.parse_known_args(argv)
+        args.args = unknown + args.args
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """Add Command args for parsing."""
+
+        parsers = parser.add_parser(cls.name, help='setup %s' % name)
+        parsers.add_argument('args', nargs='*', default=[], help='args')
+        parsers.set_defaults(command=cls)
+
+    def del_file(self, filename):
+        if os.path.exists(filename):
+            os.remove(filename)
+
+class HardCmd(SSPLResetCmd):
+    """Hard reset Cmd"""
+    name = "hard"
+    user_present=False
+    
+    def __init__(self, args):
+        super().__init__(args)
+
+    def process(self):
+        PRODUCTS=None
+        print("Processing %s %s" %(self.name, self._args))
+        if len(self._args) > 0:
+            if self._args[0] == "-p":
+                if len(self._args) > 1:
+                    product = self._args[1]
+                    if product not in enabled_products:
+                        self.usage(self.name)
+                        return
+                else:
+                    self.usage(self.name)
+                    return
+            else:
+                self.usage(self.name)
+                return
+
+        # stop sspl service   
+        CMD = "systemctl stop sspl-ll.service"
+        SSPLSetup._send_command(CMD)
```

Comment:
As @cdeshmukh ask.
use python dbus api instead of systemctl direct cmd here
refer https://stackoverflow.com/questions/33646374/starting-a-systemd-service-via-python

Reply:
Done